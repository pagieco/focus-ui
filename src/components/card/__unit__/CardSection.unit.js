import { shallowMount } from '@vue/test-utils';
import { CardSection } from '../index';

const shallowWrap = propsData => shallowMount(CardSection, { propsData });

describe('CardSection', () => {
  it('is a valid vue component', () => {
    expect(CardSection).toBeAComponent();
  });

  it('matches the snapshot', () => {
    expect(shallowMount(CardSection).element).toMatchSnapshot();
  });

  it('uses the title prop', () => {
    const title = 'Amazing Title';
    const wrapper = shallowWrap({ title });

    expect(wrapper.text()).toContain(title);
  });

  it('uses the subdued prop', () => {
    const wrapper = shallowWrap({ subdued: true });

    expect(wrapper.classes()).toContain('card__section--subdued');
  });

  it('uses the shallow prop', () => {
    const wrapper = shallowWrap({ shallow: true });

    expect(wrapper.classes()).toContain('card__section--shallow');
  });
});
