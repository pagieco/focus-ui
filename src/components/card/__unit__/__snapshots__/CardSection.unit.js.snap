// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardSection matches the snapshot 1`] = `
<div
  class="card__section"
>
  <!---->
   
</div>
`;
